src/core/socket.c

```diff
@@ -2295,6 +2366,16 @@ static int socket_serialize(Unit *u, FILE *f, FDSet *fds) {
                 }
         }
 
+        HASHMAP_FOREACH(k, s->peers_by_address, i) {
+                _cleanup_free_ char *t = NULL;
+
+                r = sockaddr_pretty(&k->peer.sa, FAMILY_ADDRESS_SIZE(k->peer.sa.sa_family), true, true, &t);
+                if (r < 0)
+                        return r;
+
+                unit_serialize_item_format(u, f, "peer", "%u %s", k->n_ref, t);
+        }
```

Comment:
I don't think we need to serialize anything here tbh. The information you are serializing here can totally be reconstructed during deserialization directly from the socket connection anyway. Just call socket_find_peer() again for the connection fds and the peer table should be reconstructued without and further need for serialization.